omod/src/main/java/org/openmrs/module/appointments/web/controller/AppointmentsController.java

```diff
@@ -70,4 +84,13 @@
         }
     }
 
+    @RequestMapping(method = RequestMethod.POST, value = "search")
+    @ResponseBody
+    public List<AppointmentDefaultResponse> search(@Valid @RequestBody AppointmentSearch appointmentSearch) {
+        List<Appointment> appointments = appointmentsService.search(appointmentSearch);
+        if(isNull(appointments)){
```

Comment:
- How can we assume that if the result is null, then start or end date is not provided??
- If start and endDate are mandatory for the POST request, then we should have validations done of the appointmentSearch object before calling the service method. 
- why annotate with @valid?  if you want to employ validation by spring, then you should appropriately annotate the model  attributes (e.g. @NotNull etc). Spring will apply the validation as per the attribute annotations and you will get appropriate validated response, without the controller code even called for execution.

Reply:
Currently all the APIs are using @Valid, so can we go ahead and merge this? And have a card in Backlog to refactor this throughout appointment scheduling?

Reply:
@valid is useless, unless you are annotating the request binding object. That should be fixed. This should not take much time. (I think I have spoken to Sowmika and told her what to do as well). Its simple spring Validator interface implementation, if the JSR validations don't kick in.

Reply:
Doing this now, when the PR contributor has most context, and when the testing would be done would be the best.